src/s390_ecc.c

```diff
@@ -759,19 +850,18 @@ unsigned int ecdsa_verify_sw(const ICA_EC_KEY *pubkey,
         goto err;
     }
 
-    if (!EC_KEY_check_key(a)) {
-        goto err;
-    }
-
     /* create ECDSA_SIG instance */
     sig = ECDSA_SIG_new();
     r = BN_bin2bn(signature, privlen, r);
     s = BN_bin2bn(signature+privlen, privlen, s);
```

Comment:
BN_bin2bn allocates BIGNUM structures for both s and r. The memory should be freed before return.

Reply:
I'm calling ECDSA_SIG_free before return and checked again what ECDSA_SIG_free does .... Here is the code from ec_asn1.c: 

void ECDSA_SIG_free(ECDSA_SIG *sig)
{
    if (sig == NULL)
        return;
    BN_clear_free(sig->r);
    BN_clear_free(sig->s);
    OPENSSL_free(sig);
}

So ECDSA_SIG_free also frees r and s.

Reply:
ok